src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -975,6 +975,21 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
             .context("`image` is not ORG/NAME:VERSION")?
             .to_string();
 
+        let container_security_context = Some(SecurityContext {
+            privileged: Some(false),
+            run_as_non_root: Some(true),
+            allow_privilege_escalation: Some(false),
+            seccomp_profile: Some(SeccompProfile {
+                type_: "RuntimeDefault".to_string(),
+                ..Default::default()
+            }),
+            capabilities: Some(Capabilities {
+                drop: Some(vec!["ALL".to_string()]),
+                ..Default::default()
+            }),
+            ..Default::default()
+        });
```

Comment:
This is by no means the original sin, and you definitely shouldn't try to address this in this PR, but having the security context stuff here _really_ makes me wish we had a better abstraction boundary between `envd` and clusters: https://github.com/MaterializeInc/cloud/issues/8141. While `envd` doesn't want to be in the business of managing StatefulSets directly at all, it _especially_ doesn't want to be in the business of having an opinion on pod security contexts.